t/t-dedup.sh

```diff
@@ -0,0 +1,83 @@
+#!/usr/bin/env bash
+
+. "$(dirname "$0")/testlib.sh"
+
+begin_test "dedup"
+(
+  set -e
+
+  reponame="dedup"
+  git init $reponame
+  cd $reponame
+
+  # Create a commit with some files tracked by git-lfs
+  git lfs track *.dat
+  echo "test data" > a.dat
+  echo "test data 2" > b.dat
+  git add .gitattributes *.dat
+  git commit -m "first commit"
+
+  # Delete file b and lock directory
+  bOid=$(git log --patch b.dat | grep "^+oid" | cut -d ":" -f 2)
+  bOid12=$(echo $bOid | cut -b 1-2)
+  bOid34=$(echo $bOid | cut -b 3-4)
+  rm ".git/lfs/objects/$bOid12/$bOid34/$bOid"
+
+  # DO
+  result=$(git lfs dedup 2>&1) && true
+  if ( echo $result | grep "This system does not support deduplication." ); then
```

Comment:
I like this approach. It makes it really easy for this to work on any system that supports it without needing to hard-code a list of systems or file systems.